vioquine/assignment1/Q2SingleLinkedList.java

```diff
@@ -0,0 +1,55 @@
+import java.util.AbstractMap;
+import java.util.Map;
+
+public class Q2SingleLinkedList {
```

Comment:
I know I said at the session yesterday that I will generally ask you to keep to your solutions but in this case I think you should try to avoid recursion. It is very expensive to keep a separate recursive call in the stack for every element in the list. If you search for the maximum recursive call stack size in java you can learn that it is by default something below 10 000. This would cause your solution to crash on an even not very exhaustive testing.

Can you think of changing your algorithm to remove recursion?

Reply:
Sure, I will implement a solution without recursion.